core/domain/classifier_services.py

```diff
@@ -100,8 +100,31 @@ def classify_string_classifier_rule(state, normalized_answer):
             training_examples.extend([
                 [doc, [str(answer_group_index)]]
                 for doc in classifier_rule_spec.inputs['training_data']])
-    if len(training_examples) > 0:
+
+    exploration = exp_services.get_exploration_by_id(exp_id)
+    algorithm_id = feconf.INTERACTION_CLASSIFIER_MAPPING['TextInput']
+    if state.classifier_model_id:
+        classifier = get_classifier_by_id(state.classifier_model_id)
+        sc_dict = classifier.cached_classifier_data
+        sc = classifier_registry.Registry.get_classifier_by_algorithm_id(
+            feconf.INTERACTION_CLASSIFIER_MAPPING['TextInput'])
+        sc.from_dict(sc_dict)
+
+    else:
+        sc = classifier_registry.Registry.get_classifier_by_algorithm_id(
+            feconf.INTERACTION_CLASSIFIER_MAPPING['TextInput'])
         sc.train(training_examples)
+        cached_classifier_data = sc.to_dict()
+        schema_version = exploration.states_schema_version
+        classifier = classifier_domain.Classifier('0', exploration.id,
+                                                  exploration.version,
```

Comment:
@anmolshkl . I used exp_services.get_exploration_by_id() method to retrieve Exploration domain object using the exploration id. The Exploration domain object has an attribute called 'version'. So, I just used 'exploration.version'. But, if exploration id is not unique and can point to different versions, then this function (get_exploration_by_id) loses its purpose right?